maths/harshad_numbers.py

```diff
@@ -0,0 +1,133 @@
+"""
+A harshad number (or more specifically an n-harshad number) is a number that's
+divisible by the sum of its digits in some given base n.
+Reference: https://en.wikipedia.org/wiki/Harshad_number
+"""
+
+
+def int_to_base(number: int, base: int) -> str:
+    digits = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    result = ""
+
+    if number < 0:
+        raise ValueError("number must be a positive integer")
+
+    while number > 0:
+        number, remainder = divmod(number, base)
+        result = digits[remainder] + result
+
+    if result == "":
+        result = "0"
+
+    return result
+
+
+def sum_of_digits(num: int, base: int) -> str:
+    """
+    Calculate the sum of digit values in a positive integer
+    converted to the given 'base'.
+    Where 'base' ranges from 2 to 36.
+
+    Examples:
+    >>> sum_of_digits(103, 12)
+    '13'
+    >>> sum_of_digits(1275, 4)
+    '30'
+    >>> sum_of_digits(6645, 2)
+    '1001'
+    >>> # bases beyond 36 and below 2 will error
+    >>> sum_of_digits(543, 1)
+    Traceback (most recent call last):
+        ...
+    ValueError: 'base' must be between 36 and 2 inclusive
+    >>> sum_of_digits(543, 37)
+    Traceback (most recent call last):
+        ...
+    ValueError: 'base' must be between 36 and 2 inclusive
+    """
+
+    if (base > 36) or (base < 2):
+        raise ValueError("'base' must be between 36 and 2 inclusive")
+
+    num_str = int_to_base(num, base)
+    res = 0
+    for char in num_str:
+        res += int(char, base)

```

Comment:
```suggestion
    res = sum(int(char, base) for char in num_str)
```
Let's avoid manually looping when possible